### PR TITLE
OCPQE-20701: add regression job for machine api

### PR DESCRIPTION
**Pull request by huali9**
For regression mapi cases on machine-api-operator repo rebases
@miyadav @sunzhaohua2 @shellyyang1989 @damdo PTAL

### CODE REVIEW

**Comment by huali9** [comment]
/pj-rehearse pull-ci-openshift-machine-api-operator-master-regression-clusterinfra-aws-ipi-mapi

File: ci-operator/config/openshift/machine-api-operator/openshift-machine-api-operator-master.yaml [hunk 1]
```diff
@@ -233,6 +237,23 @@ tests:
   steps:
     cluster_profile: openstack-vexxhost
     workflow: openshift-e2e-openstack-ipi
+- as: regression-clusterinfra-aws-ipi-mapi
+  optional: true
+  run_if_changed: ^(go\.mod|go\.sum)$
```
Comment (damdo) [inline ci-operator/config/openshift/machine-api-operator/openshift-machine-api-operator-master.yaml:242]: This is to run the job only on dependency bumps, and not on normal code changes right?

**Review by damdo — COMMENTED** [review]
Thanks @huali9
It looks like the rehearsal for the regression was able to run

So
/approve

**Comment by huali9** [comment]
> Thanks @huali9 It looks like the rehearsal for the regression was able to run
> 
> So /approve

Thank you @damdo But I am still looking into why the job failed.

File: ci-operator/config/openshift/machine-api-operator/openshift-machine-api-operator-master.yaml (hunk 1 above)
Comment (huali9) [reply ci-operator/config/openshift/machine-api-operator/openshift-machine-api-operator-master.yaml:242]: Right, I am adding this job aim to run mapi regression cases when machine-api-operator repo rebases

**Comment by huali9** [comment]
/pj-rehearse pull-ci-openshift-machine-api-operator-master-regression-clusterinfra-aws-ipi-mapi

**Pull request merged**